Project 4/README.md

```diff
@@ -45,3 +45,212 @@ You're probably thinking, "why would I submit a level 8 kata if they're not wort
+  UNION {?item wdt:P8810 ?parent.
+               ?parent wdt:P106 wd:Q82955.}     #or item's one of parents(other than father and mother defined by P22 and P25).
+  UNION {?item wdt:P3448 ?stepparent.
+               ?stepparent wdt:P106 wd:Q82955.}.  #or item's one of stepparents is a politician. Noting that the content of P25, P8810, P3448 and P106 may overlap. 
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P106 wd:Q82955.}     #or item's spouse is a politician.     
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P22 ?sFather.
+               ?sFather wdt:P106 wd:Q82955.}      #or item's spouse' father is a politician.
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P25 ?sMother.
+               ?sMother wdt:P106 wd:Q82955.}      #or item's spouse' mothe is a politician.
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P8810 ?sParent.
+               ?sParent wdt:P106 wd:Q82955.}      #or one of the item's spouse' parent is a politician.
+  UNION {?item wdt:P26 ?spouse.
+               ?spouse wdt:P3448 ?sStepparent.
+               ?sStepparent wdt:P106 wd:Q82955.}. #or one of the item's spouse's stepparent is a politician. 
+  
+  ?item wdt:P569 ?dob.
+  FILTER (?dob >= "1949-10-01"^^xsd:dateTime).   # item is borned later than Jan.1, 1949. 
+  
+  
+  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE], zh". } # Helps get the label in your language, if not, then zh language
+} 
+
+
+```
+
+Query 4. (Kata 2 - 25 points)
+
+Title: Law firms in the US
+
+Description:
+
+The company H collcets legal cases and sells the relevant databases, such as case data, case briefs collections, etc. In order to be aware the target 
+clients, the marketing manager Julia wants her secratory Jim to do the research of law firms in the US. She wants to establish a database of law firms
+(the annual income > USD.5,000,000), including the following inforamtion:  
+1) the name of the law firm.
+2) the city in which the law firm or its headquarter locates.
+3) the number of years since the law firm has founded.
+4) the number of lawyers in the law firm.
+5) the number of anual income in the last year.
+6) the popularity of the law firm. 
+
+The secratory Jim plans to use his SPARQL knowledge and Wikidata Query Service to do the research. Jim names the query "Law firms in the US."
+
+```
+#Law firms in the US
+
+PREFIX wd: <http://www.wikidata.org/entity/>
+PREFIX wdt: <http://www.wikidata.org/prop/direct/>
+PREFIX wikibase: <http://wikiba.se/ontology#>
+
+SELECT ?item ?itemLabel ?cityLabel ?yearFounded ?annualIncome2022 ?popurarity 
+WHERE {
+  ?item wdt:P31 wd:Q613142.   #item is an instance of law firm.                      
+  ?item wdt:P17 wd:Q30.       #item is a US company. 
+  
+  FILTER (!{?item wdt:P31 wd:Q163740.} UNION {?item wdt:P31 wd:Q613142}). #exclude an institute which is both a law firm and a nonprofit orginzation.
+  
+  OPTIONAL {
```

Comment:
These OPTIONAL clauses operate as conditionals. So this should read more like "if the city is in the law firm's headquarter location"